modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -81,14 +84,25 @@ const EditorUpload = function (editor: Editor): EditorUpload {
     return content;
   };
 
-  const replaceUrlInUndoStack = function (targetUrl: string, replacementUrl: string) {
-    Arr.each(editor.undoManager.data, function (level) {
+  const replaceImageUrlForPlaceholder = (content: string, targetUrl: string, replacementUrl: string): string => {
+    const placeholder = Settings.getPlaceholder(editor);
+
+    content = replaceString(content, 'src="' + targetUrl + '"', 'src="' + replacementUrl + '" data-mce-placeholder="' + placeholder + '" ');
```

Comment:
This core is almost identical to the code above, can this be refactored to remove the duplication?